src/WebJobs.Script/Description/FunctionDescriptorProvider.cs

```diff
@@ -213,6 +220,11 @@ protected internal virtual void ValidateBinding(BindingMetadata bindingMetadata)
             {
                 throw new ArgumentException($"{ScriptConstants.SystemReturnParameterBindingName} bindings must specify a direction of 'out'.");
             }
+
+            if (string.IsNullOrEmpty(bindingMetadata.Type))
```

Comment:
In writing the tests, it turned out that some of the test bindings weren't specifying a Type, causing errors. E.g. in the new trigger validations above, IsTrigger would throw for bindings that didn't specify a Type, since it relies on parsing that value.

Reply:
To be clear, this scenario already threw an exception, correct? This is just a better exception? 

Just trying to avoid something now throwing that was accidentally "working" before.

Reply:
Yes - as you can see in the code [here](https://github.com/Azure/azure-functions-host/blob/8596a27c5a3cf1c26546418fe8ac6bcf132f808b/src/WebJobs.Script.Abstractions/Description/Binding/BindingMetadata.cs#L56) for IsTrigger which is called above, we assume Type is always set. This was resulting in a null ref exception before, now a better message.